File: golang/accdemo/accmodel.go
package main

import (
	"math/rand"
	"strconv"
	"strings"
	"sync"

	"github.com/google/uuid"
)

var FoodStrs = []string{"Apple", "Banana", "Guava", "Orange", "Blackberry", "Mango", "Kiwi", "Raspberry", "Pineapple", "Avacado", "Onion", "Lettuce", "Cheese", "Almond", "Cake", "Walnut"}
var SuffixStrs = []string{"LLC", "Inc", "Corp", "Corp", "Ltd", "", "", "", "", "", "", "", ""}
var ModWordStrs = []string{"Star", "Lightning", "Flash", "Media", "Data", "Micro", "Net", "Echo", "World", "Red", "Blue", "Green", "Yellow", "Purple", "Tele", "Cloud", "Insta", "Face", "Super"}
var EmailStrs = []string{"mike", "matt", "michelle", "pat", "jim", "marc", "andrew", "alan", "henry", "jenny"}

type AccType struct {
	AccId   string
	AccName string
	IsPaid  bool
	Email   string
}

type AccModel struct {
	Lock *sync.Mutex
	Accs []*AccType
}

func MakeAccModel() *AccModel {
	rtn := &AccModel{Lock: &sync.Mutex{}}
	rtn.RegenAccounts()
	return rtn
}

func (m *AccModel) AccById(id string) *AccType {
	m.Lock.Lock()
	defer m.Lock.Unlock()
	return m.accByIdNoLock(id)
}

func (m *AccModel) CopyAccList() []*AccType {
	m.Lock.Lock()
	defer m.Lock.Unlock()
	rtn := make([]*AccType, len(m.Accs))
	for i := 0; i < len(m.Accs); i++ {
		accCopy := *m.Accs[i]
		rtn[i] = &accCopy
	}
	return rtn
}

func (m *AccModel) accByIdNoLock(id string) *AccType {
	for _, acc := range m.Accs {
		if acc.AccId == id {
			return acc
		}
	}
	return nil
}

func (m *AccModel) CreateAcc(name string, email string) string {
	m.Lock.Lock()
	defer m.Lock.Unlock()
	accId := uuid.New().String()
	m.Accs = append(m.Accs, &AccType{AccId: accId, AccName: name, Email: email})
	return accId
}

func (m *AccModel) RemoveAcc(id string) bool {
	m.Lock.Lock()
	defer m.Lock.Unlock()

	pos := -1
	for idx, acc := range m.Accs {
		if acc.AccId == id {
			pos = idx
			break
		}
	}
	if pos == -1 {
		return false
	}
	m.Accs = append(m.Accs[:pos], m.Accs[pos+1:]...)
	return true
}

func (m *AccModel) Upgrade(id string) bool {
	m.Lock.Lock()
	defer m.Lock.Unlock()
	acc := m.accByIdNoLock(id)
	if acc == nil {
		return false
	}
	acc.IsPaid = true
	return true
}

func (m *AccModel) Downgrade(id string) bool {
	m.Lock.Lock()
	defer m.Lock.Unlock()
	acc := m.accByIdNoLock(id)
	if acc == nil {
		return false
	}
	acc.IsPaid = false
	return true
}

func (m *AccModel) RegenAccounts() {
	m.Lock.Lock()
	defer m.Lock.Unlock()
	m.Accs = nil
	for i := 0; i < 5; i++ {
		m.Accs = append(m.Accs, MakeRandomAcc())
	}
}

func RandomWord(list []string) string {
	return list[rand.Int31n(int32(len(list)))]
}

func MakeRandomAcc() *AccType {
	return &AccType{
		AccId:   uuid.New().String(),
		AccName: strings.TrimSpace(RandomWord(ModWordStrs) + " " + RandomWord(FoodStrs) + " " + RandomWord(SuffixStrs)),
		IsPaid:  false,
		Email:   RandomWord(EmailStrs) + strconv.Itoa(int(rand.Int31n(70)+10)) + "@nomail.com",
	}
}

File: python/accdemo/accmodel.py
import uuid
import random

FOOD_STRS = ["Apple", "Banana", "Guava", "Orange", "Blackberry", "Mango", "Kiwi", "Raspberry", "Pineapple", "Avacado", "Onion", "Lettuce", "Cheese", "Almond", "Cake", "Walnut"]
SUFFIX_STRS = ["LLC", "Inc", "Corp", "Corp", "Ltd", "", "", "", "", "", "", "", ""]
MODWORDS_STRS = ["Star", "Lightning", "Flash", "Media", "Data", "Micro", "Net", "Echo", "World", "Red", "Blue", "Green", "Yellow", "Purple", "Tele", "Cloud", "Insta", "Face", "Super"]
EMAIL_STRS = ["mike", "matt", "michelle", "pat", "jim", "marc", "andrew", "alan", "henry", "jenny"]

class AccModel:
    def __init__(self):
        self.regen_accounts()

    def acc_by_id(self, accid):
        for acc in self.accs:
            if acc.accid == accid:
                return acc
        return None

    def all_accs(self):
        return self.accs

    def create_acc(self, name, email):
        acc = AccType(name=name, email=email)
        self.accs.append(acc)
        return acc.accid

    def remove_acc(self, accid):
        self.accs[:] = [acc for acc in self.accs if acc.accid != accid]

    def upgrade(self, accid):
        acc = self.acc_by_id(accid)
        if acc is not None:
            acc.is_paid = True

    def downgrade(self, accid):
        acc = self.acc_by_id(accid)
        if acc is not None:
            acc.is_paid = False

    def regen_accounts(self):
        self.accs = []
        for i in range(5):
            self.accs.append(make_random_acc())
    

class AccType:
    def __init__(self, name, email, is_paid=False):
        self.accid = str(uuid.uuid4())
        if name is None:
            raise ValueError("name cannot be None")
        if email is None:
            raise ValueError("email cannot be None")
        self.name = name
        self.email = email
        self.is_paid = is_paid

    def to_dict(self):
        return {"AccId": self.accid, "AccName": self.name, "Email": self.email, "IsPaid": self.is_paid}
    

def random_word(word_list):
    return word_list[random.randrange(len(word_list))]

def make_random_acc():
    name = (random_word(MODWORDS_STRS) + " " + random_word(FOOD_STRS) + " " + random_word(SUFFIX_STRS)).strip()
    email = random_word(EMAIL_STRS) + str(random.randrange(70) + 10) + "@nomail.com"
    return AccType(name=name, email=email)
